app/utils/service_client/middleware/jwt_authenticator.rb

```diff
@@ -0,0 +1,25 @@
+module ServiceClient
+  module Middleware
+
+    TOKEN_DATA_KEYS = Set.new([:marketplaceId, :actorId, :role])
+
+    class JwtAuthenticator < MiddlewareBase
+      def enter(ctx)
+        unless APP_CONFIG.harmony_api_disable_authentiation
+          token = create_token(ctx[:opts][:auth_data])
+          ctx[:req][:headers]["Authorization"] = "Token #{token}"
+        end
+        ctx
+      end
+
+      private
+
+      def create_token(auth_data)
+        if !auth_data || auth_data.keys.to_set != TOKEN_DATA_KEYS
```

Comment:
Would it make sense to make a new TokenData entity? Then you could add types (uuid for marketplace and actor and enum for role) to the entity.